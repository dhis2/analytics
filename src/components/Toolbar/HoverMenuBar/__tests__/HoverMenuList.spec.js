import { render, screen } from '@testing-library/react'
import React from 'react'
import { HoverMenuList, HoverMenuListItem } from '../index.js'

describe('<HoverMenuList/>', () => {
    const dataTest = 'test'
    const childNode = 'children'

    test('renders children', () => {
        render(<HoverMenuList>{childNode}</HoverMenuList>)
        expect(screen.getByText(childNode)).toBeInTheDocument()
    })

    test('accept a `className` prop', () => {
        const className = 'className'

        render(
            <HoverMenuList dataTest={dataTest} className={className}>
                {childNode}
            </HoverMenuList>
        )

        expect(screen.getByTestId(dataTest)).toHaveClass(className)
    })

    test('accepts a `dataTest` prop', () => {
        render(<HoverMenuList dataTest={dataTest}>{childNode}</HoverMenuList>)

        expect(screen.getByTestId(dataTest)).toBeInTheDocument()
    })

    test('accept a `dense` prop', () => {
        render(
            <HoverMenuList dense>
                <HoverMenuListItem label="item 1" />
                <HoverMenuListItem label="item 2" />
            </HoverMenuList>
        )

        expect(screen.getByText('item 1').closest('li')).toHaveClass('dense')
        expect(screen.getByText('item 2').closest('li')).toHaveClass('dense')
    })

    test('accept a `maxHeight` prop', () => {
        const maxHeight = '100000px'

        const { container } = render(
            <HoverMenuList dataTest={dataTest} maxHeight={maxHeight}>
                {childNode}
            </HoverMenuList>
        )

        expect(container).toMatchSnapshot()
    })

    test('accept a `maxWidth` prop', () => {
        const maxWidth = '100000px'

        const { container } = render(
            <HoverMenuList dataTest={dataTest} maxWidth={maxWidth}>
                {childNode}
            </HoverMenuList>
        )
        expect(container).toMatchSnapshot()
    })
})
