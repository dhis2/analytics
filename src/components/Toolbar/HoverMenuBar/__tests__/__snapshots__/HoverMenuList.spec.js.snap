// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HoverMenuList/> accept a \`maxHeight\` prop 1`] = `
<div>
  <ul
    data-test="test"
  >
    children
    <style>
      
                    ul {
                        position: relative;
                        margin: 0;
                        padding: 0;
                        user-select: none;
                        background: #ffffff;
                        border: 1px solid #f3f5f7;
                        border-radius: 3px;
                        box-shadow: 0px 10px 15px -3px rgba(33,41,52,0.1), 0px 4px 6px -2px rgba(33,41,52,0.05);
                        display: inline-block;
                        min-width: 128px;
                        max-width: 380px;
                        max-height: 100000px;
                        padding: 4px 0;
                        overflow: auto;
                        list-style: none;
                    }
                
    </style>
  </ul>
</div>
`;

exports[`<HoverMenuList/> accept a \`maxWidth\` prop 1`] = `
<div>
  <ul
    data-test="test"
  >
    children
    <style>
      
                    ul {
                        position: relative;
                        margin: 0;
                        padding: 0;
                        user-select: none;
                        background: #ffffff;
                        border: 1px solid #f3f5f7;
                        border-radius: 3px;
                        box-shadow: 0px 10px 15px -3px rgba(33,41,52,0.1), 0px 4px 6px -2px rgba(33,41,52,0.05);
                        display: inline-block;
                        min-width: 128px;
                        max-width: 100000px;
                        max-height: auto;
                        padding: 4px 0;
                        overflow: auto;
                        list-style: none;
                    }
                
    </style>
  </ul>
</div>
`;
